### DISCUSSION TITLE
terminal emulation issue - web interface and S-132 with cp/m games

### DISCUSSION THREAD

**Discussion by inerlogic** [thehighnibble/thehighnibble.github.io#104] [S-132 add-on for IMSAI & Cromemco replicas]
hello all,
i downloaded "nemesis" from retroarchive.org [http://www.retroarchive.org/cpm/games/games.htm](url) it contains "termdef" a program to setup up the game for your particular terminal. DEC VT-52/VT100 is an available option, but on choosing that option, the S-132 doesn't clear the screen when printing the terminal test, or when printing centered text menu options. the web interface's tty doesn't even try to be correct (see attached)

terminal options menu
![01terminal options](https://github.com/thehighnibble/thehighnibble.github.io/assets/3458154/9318b0a3-e50a-420c-921e-f5b230886b1c)

terminal settings in TERMDEF
![03terminal settings](https://github.com/thehighnibble/thehighnibble.github.io/assets/3458154/68901af7-0e0b-4e4a-a4d0-7909df21bccc)

screen shot of the test screen from the web tty
![02web interface tty](https://github.com/thehighnibble/thehighnibble.github.io/assets/3458154/058a8f73-32ed-400b-97ee-4791a8c36966)

the same screen through the S-132, the menu should have cleared and the test text be present
![03testsettings s132](https://github.com/thehighnibble/thehighnibble.github.io/assets/3458154/ec33c7e0-e066-43bc-bd80-25b57950be9d)

vertical text ( 0 human 1 elf, etc) screen should have cleared
![04s132-no clear-screen](https://github.com/thehighnibble/thehighnibble.github.io/assets/3458154/f7b7ae12-9399-48f7-9690-7bdd8883ff23)

the same screen on the web tty with the text jumbled
![04webtty_should be vertical](https://github.com/thehighnibble/thehighnibble.github.io/assets/3458154/e1cbe91e-25d4-4299-ba14-02e93d1182b3)

**Comment by nbreeden2** [comment]
The  "home-and-clear " <esc> sequence you show isn't one I recognize for the VT100. It may be valid on a later terminal such as the VT220.

These are the sequencies I use and are the ones I tested on the VT132

Clear screen:   esc[2J
Home cursor:  esc[H

In HEX:
1BH 5BH 32H 4AH
1BH 5BH 48H 

Set the sequence like this:

1BH 5BH 32H 4AH 1BH 5BH 48H 

Reference:
https://espterm.github.io/docs/VT100%20escape%20codes.html
https://www.asciitable.com/

**Reply by inerlogic** [reply]
Sitting here with the escape codes and an ascii chart, still no luck.
![20231004_183218](https://github.com/thehighnibble/thehighnibble.github.io/assets/3458154/f39eaa38-8022-496c-a298-088baf6ed726)
![20231004_183255](https://github.com/thehighnibble/thehighnibble.github.io/assets/3458154/a4dc0ca8-b645-4a02-b216-d26a63baa1f0)
